### PR TITLE
#188 Cancel an on-going download

### PR DESCRIPTION
**Pull request by tuancoltech**

### CODE REVIEW

File: app/src/main/java/ai/elimu/appstore/synchronization/AppListAdapter.java [hunk 1]
```diff
@@ -107,12 +117,12 @@ public void onBindViewHolder(final ViewHolder holder, final int position) {
         holder.progressBarDownload.setProgress(downloadStatus.getDownloadProgress());
 
         if (downloadStatus.isDownloading()) {
-            holder.progressBarDownload.setVisibility(View.VISIBLE);
-            holder.textDownloadProgress.setVisibility(View.VISIBLE);
+            holder.layoutProgress.setVisibility(View.VISIBLE);
+            holder.ivCancelDownload.setVisibility(View.VISIBLE);
```
Comment (jo-elimuai) [inline app/src/main/java/ai/elimu/appstore/synchronization/AppListAdapter.java]: What does `iv` mean?

File: app/src/main/java/ai/elimu/appstore/synchronization/AppListAdapter.java [hunk 2]
```diff
@@ -337,12 +347,14 @@ public void run() {
                     Call<ResponseBody> call = downloadApplicationService.downloadApplicationFile(getFileUrl
                             (applicationVersion));
 
+                    appDownloadRequests.set(position, call);
```
Comment (jo-elimuai) [inline app/src/main/java/ai/elimu/appstore/synchronization/AppListAdapter.java]: What is happening here? Can you add a comment explaining the functionality?

**Comment by jo-elimuai** [comment]
The "Invalid checksum" Toast should not appear when a download is cancelled.

File: app/src/main/java/ai/elimu/appstore/synchronization/AppListAdapter.java (hunk 1 above)
Comment (tuancoltech) [reply app/src/main/java/ai/elimu/appstore/synchronization/AppListAdapter.java]: It stands for ImageView.

Comment (jo-elimuai) [reply app/src/main/java/ai/elimu/appstore/synchronization/AppListAdapter.java]: Kan you rename to make it less confusing?

File: app/src/main/res/layout/activity_app_list_item.xml [hunk 3]
```diff
@@ -56,18 +57,34 @@
...
+                android:progress="30" />
+
+            <TextView
+                android:id="@+id/textViewDownloadProgress"
+                android:layout_width="wrap_content"
+                android:layout_height="wrap_content" />
+
+        </LinearLayout>
+
+        <ImageView
```
Comment (jo-elimuai) [inline app/src/main/res/layout/activity_app_list_item.xml]: Shouldn't this be a `Button` or `ImageButton` instead of an image?

**Pull request closed without merging**